### PR TITLE
Wait for the media fetch

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier

Describe your changes:
Branched off of #552. This fixes an issue where sometimes the reference to the media would not resolve because the request for it hadn't finished before the tutorial modal opened up. We now wait for the request for the media to finish before setting the tutorial and calling to show the modal. 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/lib-classifier/src/store/TutorialStore.js [hunk 1]
```diff
@@ -140,14 +135,19 @@ const TutorialStore = types
...
       const tutorialsClient = getRoot(self).client.tutorials
+      const upp = getRoot(self).userProjectPreferences
 
       self.loadingState = asyncStates.loading
       try {
         const response = yield tutorialsClient.get({ workflowId: workflow.id })
         const { tutorials } = response.body
         if (tutorials && tutorials.length > 0) {
-          tutorials.forEach(tutorial => self.fetchMedia(tutorial))
+          yield Promise.all(tutorials.map(tutorial => self.fetchMedia(tutorial)))
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/TutorialStore.js]: I don't think `fetchMedia` is called anywhere else in the code so it can be a private method.
```suggestion
          yield Promise.all(tutorials.map(tutorial => flow(fetchMedia)(tutorial)))
```

**Comment by eatyourgreens** [comment]
The code works when I test this in a browser, so it looks good. However, the tests fail if `fetchMedia` isn't stubbed, which makes me worried.

**Comment by srallen** [comment]
They were failing because the panoptes client isn't being stubbed to return a mocked medium resource for those get requests for those tests. We now have a `yield` statement on those requests and it'll wait for a promise to resolve or reject. If we aren't returning a promise from `fetchMedia` then the test will fail. It was easier to just stub `fetchMedia` there to just return a promise to get the tests passing then to make what I think is a hard to follow mocking of the client even more complicated. `fetchMedia` itself is being tested too, so imo, we aren't missing test coverage here.

**Comment by eatyourgreens** [comment]
Weird, because I was seeing a media resource coming back from the mocked client, but tests were failing anyway.

**Comment by srallen** [comment]
It's being mocked in a different describe block I think, but I could be wrong about that... this is what I mean, it's confusing/hard to follow how the panoptes client is being mocked.

**Comment by eatyourgreens** [comment]
Yeah. This makes me nervous because `fetchMedia` is only being exported in order to pass the tests, and I'm not 100% sure that the tests are actually testing the code the way that we think.

My thinking was that as long as `tutorials.getAttachedImages()` is stubbed in the tests, then they should still work. As far as I can tell, it is stubbed and resolves with a mock image, but the tests fail if we actually run that stub via the `fetchMedia` call.

**Comment by eatyourgreens** [comment]
Nevermind, I completely missed that `Promise.all` isn't appropriate here. I'll open a PR.

**Comment by eatyourgreens** [comment]
I can't figure out why `Promise.all` breaks the tests, unless `fetchMedia` is stubbed with `Promise.resolve()`, but the behaviour I'm seeing is that the expectations run before `setTutorial` has run, or before `showTutorialInModal` has run. Look at #601 for an example.

**Comment by srallen** [comment]
I looked it over and the last two tests for the describe block for `Actions > showTutorialInModal` didn't actually need fetchMedia's stub calling a fake function to return the promise. It is being stubbed correctly in the client for those two tests. It's the first test that didn't and needs the `callsFake` because `getAttachedImages` is not being stubbed.

**Comment by srallen** [comment]
I think it's the confusing way I'm mocking panoptes.js. I'd like to revisit that and get a standard way that this done and a part of panoptes.js itself.

**Comment by eatyourgreens** [comment]
Tests still pass without this change, so there's a test missing to catch the original bug.
